test/e2e/useraccount_test.go

```diff
@@ -0,0 +1,157 @@
+package e2e
+
+import (
+	"context"
+	"testing"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	"github.com/codeready-toolchain/member-operator/pkg/apis"
+	"github.com/codeready-toolchain/member-operator/pkg/controller/useraccount"
+	userv1 "github.com/openshift/api/user/v1"
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/operator-framework/operator-sdk/pkg/test/e2eutil"
+	uuid "github.com/satori/go.uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func TestUserAccount(t *testing.T) {
+	userAccList := &toolchainv1alpha1.UserAccountList{}
+
+	err := framework.AddToFrameworkScheme(apis.AddToScheme, userAccList)
+	require.NoError(t, err, "failed to add custom resource scheme to framework")
+
+	ctx := framework.NewTestCtx(t)
+	defer ctx.Cleanup()
+
+	err = ctx.InitializeClusterResources(&framework.CleanupOptions{TestContext: ctx, Timeout: cleanupTimeout, RetryInterval: cleanupRetryInterval})
+	require.NoError(t, err, "failed to initialize cluster resources")
+	t.Log("Initialized cluster resources")
+
+	namespace, err := ctx.GetNamespace()
+	require.NoError(t, err, "failed to get namespace where operator needs to run")
+
+	// get global framework variables
+	f := framework.Global
+	client := f.Client.Client
+
+	// wait for operator to be ready
+	err = e2eutil.WaitForDeployment(t, f.KubeClient, namespace, "member-operator", 1, retryInterval, timeout)
+	require.NoError(t, err, "failed while waiting for operator deployment")
+
+	t.Log("member-operator is ready and running state")
+
+	extraUserAcc := createUserAccount(t, f, ctx, "amar")
+	t.Log("extra useraccount created at start")
+
+	// create useraccount
+	userAcc := newUserAcc(t, f, ctx, "johnsmith")
+	err = f.Client.Create(context.TODO(), userAcc, &framework.CleanupOptions{TestContext: ctx, Timeout: cleanupTimeout, RetryInterval: cleanupRetryInterval})
+	require.NoError(t, err)
+	t.Logf("user account '%s' created", userAcc.Name)
+
+	t.Run("verify_useraccount_ok", func(t *testing.T) {
```

Comment:
just a side note - the first parameter is a string that can contain spaces - it's not necessary to put underscores there now keep it short - it can contain a longer description for better understanding.

Reply:
thanks for info.